pkg/writer/writer.go

```diff
@@ -1,56 +1,98 @@
 package writer
 
 import (
-	"errors"
 	"fmt"
 	"io"
+	"os"
+	"sync"
 
+	"github.com/bom-squad/protobom/pkg/formats"
+	"github.com/bom-squad/protobom/pkg/native"
+	drivers "github.com/bom-squad/protobom/pkg/native/serializers"
 	"github.com/bom-squad/protobom/pkg/sbom"
-	"github.com/bom-squad/protobom/pkg/writer/options"
 )
 
-type Option func(*Writer)
+type Writer struct {
+	Indent int
+	Format formats.Format
+}
 
-func New() *Writer {
-	return &Writer{
-		impl:    &defaultWriterImplementation{},
-		Options: options.Default,
+var (
+	regMtx       sync.RWMutex
+	serializers  = make(map[formats.Format]native.Serializer)
+	defaultIdent = 4
+)
+
+func New(opts ...WriterOption) *Writer {
+	r := &Writer{
+		Indent: defaultIdent,
```

Comment:
Should we have a default format we want to write to?

Reply:
(I added one)

Reply:
Could be, but the question is reacting to a problem. Right now the format is an option in the writer but as we've been seeing when using the library in production, I think this was a design flaw. My feeling now is that you should be able to command, on each run, which format to use when writing the documents. Something equivalent to "_write this doc in format x to this stream_", like this:

`writer.WriteStream(doc, format, r io.Writer)`

Unfortunately, this would break the API. To bridge it we could still implement a `write.WriteFormatToStream()` or something

Reply:
(Good morning @puerco :) ) 

I feel perfectly fine with breaking this now, I agree with your suggestion, makes a lot of sense to avoid creating multiple instances of a writer. also if someone would want to wrap this with "helpers" and implement 
```go
w := writerhelper.wrap(writer.New())
w.WriteCDX15Stream(doc, wr)
...

it wouldn't be possible without allowing someone to change the format you serialize to dynamically

Reply:
If ever there were a time to break the API, now is the time!  My suggestion would be to make it right now for the long term, and not worry so much about breaking the API in the immediate term before we have significant usage.